src/test/mod.rs

```diff
@@ -52,7 +52,7 @@ impl Test {
         }
     }
 
-    pub fn handle_key(&mut self, key: KeyEvent) {
+    pub fn handle_key(&mut self, key: KeyEvent, backspace: bool) {
```

Comment:
I would prefer to store the word backtracking configuration in the `Test` struct so it remains an (almost) pure state machine with keypresses as inputs.

Reply:
Currently didn't know how to implement this and went with the old approach and with the other changes you suggested.

Reply:
No worries, I can make the change.